src/Synth/OscilGen.h

```diff
@@ -170,10 +199,10 @@ class OscilGen:public Presets
             oldmodulationpar3;
 
 
-        fft_t *basefuncFFTfreqs; //Base Function Frequencies
+        std::vector<fft_t> basefuncFFTfreqs; //Base Function Frequencies
```

Comment:
I don't know how to attach a comment to a whole commit in this interface, but IMO this commit seems unnecessary. For the later commits, all that is needed is harmonic content given wave parameter=x, right?

Would it make sense to have a simple get_harmonics_with_param_equal_x(x)? Perhaps I'm oversimplying things.

Reply:
I thought it's required because the `PADnoteParameters::sample` vectors can be calculated by different threads for the different frequencies (PAD synth multithreading - "If one thread wants to use wave parameter x, the other one can't use wave parameter y"). Most parts of `OscilGen::get` look easy to make thread-safe, but a call to `prepare` seems always needed due to the new basefunc parameter (`needPrepare` returns true if that one is changed), and I guess making `prepare` thread safe is as bad as copying the whole class?